### PR TITLE
fix: change reporting view name to avoid conflict when upgrade from 1.0.x

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [x] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [x] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/common/constants-ln.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+constant.ts
```
Comment (zxkane) [inline src/common/constants-ln.ts:1]: Why did you create this link file?

File: src/analytics/lambdas/custom-resource/create-schemas.ts [hunk 2]
```diff
@@ -386,11 +387,12 @@ async function createViewForReporting(props: ResourcePropertiesType, biUser: str
       table_event_parameter: odsTableNames.event_parameter,
       table_user: odsTableNames.user,
       table_item: odsTableNames.item,
+      viewVersion: REPORTING_VIEW_VERSION,
       ...SQL_TEMPLATE_PARAMETER,
     };
 
     for (const viewDef of props.reportingViewsDef) {
-      views.push(viewDef.sqlFile.replace('.sql', ''));
+      views.push(viewDef.sqlFile.replace('.sql', '') + REPORTING_VIEW_VERSION);
```
Comment (zxkane) [inline src/analytics/lambdas/custom-resource/create-schemas.ts:395]: In the next version, just a few SQLs are changed. What do we do?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 2 above)
Comment (llmin) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:395]: update all views version

File: src/common/constants-ln.ts (hunk 1 above)
Comment (llmin) [reply src/common/constants-ln.ts:1]: resolve the path conflict in dashboard.ts,  this file is used both in reporting and data modeling.

Comment (zxkane) [reply src/common/constants-ln.ts:1]: I just see `dashboard.ts` is referred to in the backend. What's the problem that you met with?

Comment (llmin) [reply src/common/constants-ln.ts:1]: import { REPORTING_VIEW_VERSION } from '../../common/constants-ln';

Comment (zxkane) [reply src/common/constants-ln.ts:1]: I don't see the reason why we need another link file on the same path.

Maybe you can extract those constants from the dashboard and save them to the constant file.

File: src/analytics/lambdas/custom-resource/create-schemas.ts (hunk 2 above)
Comment (zxkane) [reply src/analytics/lambdas/custom-resource/create-schemas.ts:395]: I'm against this design to introduce lots of unnecessary new MVs for every upgrade. I prefer to directly add the initial version to the file name to identify the new or changed MVs.

File: src/analytics/private/sqls/redshift/dashboard/clickstream_device_view_v1.sql [hunk 3]
```diff
@@ -1,4 +1,4 @@
-CREATE MATERIALIZED VIEW {{schema}}.clickstream_device_view 
+CREATE MATERIALIZED VIEW {{schema}}.clickstream_device_view_v1
```
Comment (zxkane) [inline src/analytics/private/sqls/redshift/dashboard/clickstream_device_view_v1.sql:1]: How about changing it to a placeholder and then defining the name in SQLDef?

File: src/analytics/private/sql-def.ts [hunk 4]
```diff
@@ -12,51 +12,44 @@
  */
 
 import { SQLDef } from './model';
+import { CLICKSTREAM_DEVICE_VIEW_NAME, CLICKSTREAM_EVENT_PARAMETER_VIEW_NAME, CLICKSTREAM_EVENT_VIEW_NAME, CLICKSTREAM_LIFECYCLE_DAILY_VIEW_NAME, CLICKSTREAM_LIFECYCLE_WEEKLY_VIEW_NAME, CLICKSTREAM_RETENTION_VIEW_NAME, CLICKSTREAM_SESSION_VIEW_NAME, CLICKSTREAM_USER_ATTR_VIEW_NAME, CLICKSTREAM_USER_DIM_VIEW_NAME } from '../../common/constant';
 
 export const reportingViewsDef: SQLDef[] = [
   {
     updatable: 'false',
-    sqlFile: 'clickstream_event_view.sql',
+    sqlFile: `${CLICKSTREAM_EVENT_VIEW_NAME}.sql`,
```
Comment (zxkane) [inline src/analytics/private/sql-def.ts:20]: We can add `viewName` instead of the file name.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/analytics/private/sql-def.ts (hunk 4 above)
Comment (llmin) [reply src/analytics/private/sql-def.ts:20]: replace sqlFile?

Comment (llmin) [reply src/analytics/private/sql-def.ts]: add viewName property

**Review by zxkane — COMMENTED** [review]
depends on #415

**Review by zxkane — APPROVED** [review]

**Review by amliuyong — APPROVED** [review]

**Pull request merged**